homeassistant/components/lyric/climate.py

```diff
@@ -318,7 +318,28 @@ async def async_set_temperature(self, **kwargs: Any) -> None:
         target_temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
         target_temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
 
-        if device.changeableValues.autoChangeoverActive:
+        if device.changeableValues.mode == LYRIC_HVAC_MODE_HEAT_COOL:
+            # If the device supports "Auto" mode, don't pass the mode when setting the
+            # temperature
+            if target_temp_low is None or target_temp_high is None:
+                raise HomeAssistantError(
+                    "Could not find target_temp_low and/or target_temp_high in"
+                    " arguments"
+                )
+            _LOGGER.debug("Set temperature: %s - %s", target_temp_low, target_temp_high)
+            try:
+                await self._update_thermostat(
+                    self.location,
+                    device,
+                    coolSetpoint=target_temp_high,
+                    heatSetpoint=target_temp_low,
+                )
+            except LYRIC_EXCEPTIONS as exception:
+                _LOGGER.error(exception)
+            await self.coordinator.async_refresh()
+        elif device.changeableValues.autoChangeoverActive:
+            # If the device uses "autoChangeoverActive", pass the "heatCoolMode" as the
+            # mode
```

Comment:
May you can try to avoid duplication of code. As far as I can see only the `mode` parameter`differs. Perhaps make a variable for the mode instead of copying code?